### PR TITLE
Iteration 3 GUI

### LINKED ISSUE #13: Register Screen, registering with used username.
Registering with a used Username rejects with "Wrong username or Password" message. Not "Username Taken".

### LINKED ISSUE #14: Register Screen, email uniqueness.
It is possible to register with multiples of the same email address.

### PR DESCRIPTION
**Pull request by JADGardner**
### TP005 - Viewing and Updating Account Information
**User Stories Complete:** 006, 007
**Authors:** James Gardner and Stijn Marynissen

Closes #13 
Closes #14

### CODE REVIEW

**Comment by JADGardner** [comment]
I've made a fair few changes to some of the classes like Launcher and ViewFactory so might be worth leaving the merge of this pull request till last to reduce any conflicts it might have with other pull requests.

**Comment by dmb537** [comment]
Yeah leaving this until last seems sensible - poke me again then :)

**Review by ojhstill — APPROVED** [review]

File: client/src/main/java/application/Launcher.java [hunk 1]
```diff
@@ -1,46 +1,55 @@
...
+   * under resources/logs. This should be passed around the program so that all classes
+   * use the same Logger. */
+  private static final Logger log = LoggerFactory.getLogger(Launcher.class);
+
   @Override
   public void start(Stage stage) throws Exception {
 
     /* Creates the connection between the Client and the Server. The program must
-      * have this connection in order to proceed. */
-    // ########### This could do with some error handling ############################
```
Comment (cmckirgan) [inline client/src/main/java/application/Launcher.java:32]: In the future could stuff like this be done as //TODO - XYZ. It just makes it easier to track

**Review by cmckirgan — APPROVED** [review]
Other than that it looks good, everyone should roll the logging out into their stories + refactoring

**Pull request merged**